smartsim/_core/launcher/step/alpsStep.py

```diff
@@ -49,13 +49,29 @@ def __init__(self, name: str, cwd: str, run_settings: AprunSettings) -> None:
         :type run_settings: AprunSettings
         """
         super().__init__(name, cwd, run_settings)
-        self.alloc = None
+        self.alloc: t.Optional[str] = None
         if not self.run_settings.in_batch:
             self._set_alloc()
 
     @property
-    def run_settings(self) -> AprunSettings:
-        return self.step_settings
+    def run_settings(self) -> RunSettings:
+        if isinstance(self.step_settings, RunSettings):
+            return self.step_settings
+        raise TypeError("Run settings must be of type RunSettings")
+
+    def _aprun_settings(self, ignore: bool = False) -> t.Optional[AprunSettings]:
+        if isinstance(self.step_settings, AprunSettings):
+            return self.step_settings
+        if not ignore:
+            raise TypeError("Run settings must be of type AprunSettings")
+        return None
+
+    def _get_mpmd(self) -> t.List[RunSettings]:
+        """temporary convenience function to return a typed list
```

Comment:
Can you capitalize the `t` in `temporary`? I think it is our convention for docstrings. This applies to similar functions added in this PR.